tuf/repository_lib.py

```diff
@@ -452,71 +464,68 @@ def _delete_obsolete_metadata(metadata_directory, snapshot_metadata,
           os.remove(metadata_path)
 
-        # Delete outdated consistent snapshots.  snapshot metadata includes
-        # the file extension of roles.
-        if consistent_snapshot and embeded_digest is not None:
+        # Delete outdated consistent snapshots.  Snapshot metadata includes the
+        # file extension of roles.  TODO: Should we leave it up to integrators
+        # to remove outdated consistent snapshots?
+        """ 
+        if consistent_snapshot and embeded_version_number is not None:
           file_hashes = list(snapshot_metadata['meta'][metadata_name_extension] \
                                         ['hashes'].values())
           if embeded_digest not in file_hashes:
             logger.info('Removing outdated metadata: ' + repr(metadata_path))
             os.remove(metadata_path)
+        """
 
 
 
 
 
-def _get_written_metadata_and_digests(metadata_signable):
+def _get_written_metadata(metadata_signable):
   """
-  Non-public function that returns the actual content of written metadata and
-  its digest.
+  Non-public function that returns the actual content of written metadata.
   """
 
-  # Explicitly specify the JSON separators for Python 2 + 3 consistent.
+  # Explicitly specify the JSON separators for Python 2 + 3 consistency.
   written_metadata_content = \
     json.dumps(metadata_signable, indent=1, separators=(',', ': '),
                sort_keys=True).encode('utf-8')
   
-  written_metadata_digests = {}
-
-  for hash_algorithm in tuf.conf.REPOSITORY_HASH_ALGORITHMS:
-    digest_object = tuf.hash.digest(hash_algorithm)
-    digest_object.update(written_metadata_content)
-    written_metadata_digests.update({hash_algorithm: digest_object.hexdigest()})
-  
-  return written_metadata_content, written_metadata_digests
+  return written_metadata_content
 
 
 
 
 
-def _strip_consistent_snapshot_digest(metadata_filename, consistent_snapshot):
+def _strip_consistent_snapshot_version_number(metadata_filename,
+                                              consistent_snapshot):
   """
-  Strip from 'metadata_filename' any digest data (in the expected
-  '{dirname}/digest.filename' format) that it may contain, and return it.
+  Strip from 'metadata_filename' any version data (in the expected
+  '{dirname}/version_number.filename' format) that it may contain, and return
+  the stripped filename and its version number as a tuple.
   """
  
-  embeded_digest = ''
+  embeded_version_number = ''
 
-  # Strip the digest if 'consistent_snapshot' is True.
-  # Example:  'targets/unclaimed/13df98ab0.django.json'  -->
+  # Strip the version number if 'consistent_snapshot' is True.
+  # Example:  'targets/unclaimed/10.django.json'  -->
   # 'targets/unclaimed/django.json'
   if consistent_snapshot:
     dirname, basename = os.path.split(metadata_filename)
-    embeded_digest = basename[:basename.find('.')]
+    embeded_version_number = basename[:basename.find('.')]
```

Comment:
Switched to using split() instead.  I don't quite enjoy reading split(".", 1) <-- The second argument might not be obvious to newcomers / inexperienced Python programmers), but accessing list elements in `basename[:basename.find('.')` is harder to read in this case.